Pr pmw3901 fixes

RomanBapst opened the pull request:
- fix wrong flow signs
- get optical flow limits from params and publish them in flow topic for the ekf

RomanBapst commented:
@DanielePettenuzzo FYI

mhkabir commented:
The sensor specs suggest that it can measure upto a maximum of 7.4 rad/s. You might want to update the FLOW_MAXR to have a higher default in the crazyflie startup script.

mhkabir commented:
http://www.pixart.com.tw/product_data.asp?product_id=168&productclassify_id=19&productclassify2_id=53&productclassify_name=Optical%20Motion%20Tracking&productclassify2_name=Optical%20Flow&partnumber=PMW3901MB-TXQT

LorenzMeier commented on src/drivers/pmw3901/pmw3901.cpp:
```diff
@@ -341,6 +344,44 @@ PMW3901::init()
+		_sensor_max_range = val;
+	}
+
+	param_t ratemax = param_find("SENS_FLOW_MAXR");
```
This is a fixed quantum for this specific sensor - why do we need to load this from a parameter?

mhkabir replied:
This was variable for the PX4Flow, depending on the lens used. Which is why it's historically a parameter.

RomanBapst replied:
@LorenzMeier @mhkabir  I will test the sensor with a value close to the max indicated rate (to give some room for control error and disturbances).

LorenzMeier replied:
@RomanBapst Could you please wrap this up tomorrow? In general, it would be good to not keep PRs hanging open. Thanks!

RomanBapst replied:
Maybe another reason why this parameter should stay configurable: At the moment the controller will limit the horizontal velocity to comply with the maximum rotation speed of the flow sensor. However, during braking maneuvers you also need to take into account the angular velocity of the vehicle which can be considerable.

LorenzMeier replied:
But for this sensor it is still fixed. I understand the intent, but I want to specifically eliminate magic tuning parameters. There is a max rate param you can use together with the P term of the controller to estimate a likely max rotation and add it to the limit. Plus I’m not sure if a short-term flow loss is a problem on braking.

At any rate, it does not belong into the driver at all.

RomanBapst commented:
@DanielePettenuzzo I cleaned up the rotation parameter. I think the only thing missing now is your PR that takes care of the bus, right?

DanielePettenuzzo commented:
@RomanBapst let's discuss this tomorrow because we would need to use different SPI busses on different boards so either we start by adding only fmu-v5 or we need to add a macro in the board config to specify the optical flow spi bus for each board.

DanielePettenuzzo commented on src/drivers/pmw3901/pmw3901.cpp:
```diff
@@ -844,17 +892,16 @@ int
-	uint8_t rotation = distance_sensor_s::ROTATION_DOWNWARD_FACING;
 
-	if (argc < 2) {
+	if (argc < 1) {
```
@RomanBapst for this driver we don't have the px4 argument check and I think that with arg = 1 we would get a segfault because argv[myoptind] is null. I will double check this tomorrow. 
What I mean is if I just write pmw3901 in the console with no command.

RomanBapst replied:
@DanielePettenuzzo Ah ok, good catch.

RomanBapst replied:
Done

mhkabir commented:
@RomanBapst @DanielePettenuzzo where does this stand? Is this ready to merge?

RomanBapst commented:
@mhkabir I updated it just now. If you agree how we implemented the SPI related defines in the board_config.h then I can go ahead and add support for the driver to the other targets. At the moment it supports crazyflie and fmuv5.
@bkueng Could you have a look at it please?

bkueng commented on src/drivers/boards/crazyflie/board_config.h:
```diff
@@ -122,6 +122,10 @@
 #define PX4_SPIDEV_EXPANSION_2      			PX4_MK_SPI_SEL(PX4_SPI_BUS_EXPANSION, 1) 		// OPTICAL FLOW BREAKOUT
 #define PX4_SPIDEV_EXPANSION_3      			PX4_MK_SPI_SEL(PX4_SPI_BUS_EXPANSION, 2)
 
+/* Btcraze optical flow deck */
```
typo: Btcraze

bkueng commented on src/drivers/pmw3901/pmw3901.cpp:
```diff
@@ -77,18 +77,6 @@
-#endif
-
-#ifdef PX4_SPIDEV_EXPANSION_2
-#define PMW3901_SPIDEV PX4_SPIDEV_EXPANSION_2
```
Is there a sensible default we can use? I'd like to avoid adding `PX4_SPI_BUS_BITCRAZE_FLOW` to all board configs that want to support the driver.
If there is none, we should add something like this:
```
#ifndef PX4_SPI_BUS_BITCRAZE_FLOW
#error "board_config.h needs to define PX4_SPI_BUS_BITCRAZE_FLOW"
#endif
```

dagar commented:
Needs a rebase. Is this otherwise working (crazyflie flow deck on a pixhawk 4)?
